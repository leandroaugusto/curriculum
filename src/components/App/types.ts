export interface AppProps {
  readonly children: React.ReactNode;
  readonly printLabel: string;
  readonly shareLabel: string;
  readonly shareTitle: string;
  readonly shareEmailBody: string;
}
