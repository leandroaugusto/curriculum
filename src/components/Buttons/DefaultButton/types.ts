export interface ButtonProps {
  handleClick?: () => void;
  readonly label: string;
  readonly className?: string;
  readonly id: string;
}
