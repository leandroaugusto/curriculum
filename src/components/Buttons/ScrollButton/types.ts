export interface ScrollButtonProps {
  readonly label: string;
}
