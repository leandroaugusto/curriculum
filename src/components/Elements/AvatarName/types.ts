export interface AvatarNameProps {
  readonly actionsLabel: string;
  readonly role: string;
}
