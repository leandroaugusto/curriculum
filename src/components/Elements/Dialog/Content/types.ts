export interface DialogContentProps {
  readonly children: React.ReactNode;
}
