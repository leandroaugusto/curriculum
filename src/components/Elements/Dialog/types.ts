export interface DialogProps {
  readonly open: boolean;
  readonly shareTitle: string;
  readonly shareEmailBody: string;
}
