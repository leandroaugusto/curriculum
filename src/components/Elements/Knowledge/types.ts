export interface KnowledgeProps {
  readonly name: string;
  readonly value: number;
}
