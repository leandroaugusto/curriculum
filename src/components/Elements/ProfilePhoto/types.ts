export interface ProfilePhotoProps {
  url?: string;
}
