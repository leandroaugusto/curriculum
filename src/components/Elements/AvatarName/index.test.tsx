import { render, screen, waitFor } from "@testing-library/react";
import { AvatarName } from ".";
import { AvatarNameProps } from "./types";

const MOCK_DATA = {
  bio: "bio",
  name: "Leandro Costa",
};

global.fetch = jest.fn(() =>
  Promise.resolve({
    json: () => Promise.resolve(MOCK_DATA),
  })
) as jest.Mock;

async function asyncComponent(
  Component: React.FC<AvatarNameProps>,
  props: AvatarNameProps
): Promise<any> {
  const ComponentResolved = await Component(props);
  return () => ComponentResolved;
}

describe("AvatarName", () => {
  it("renders the AvatarName element", async () => {
    const AvatarResolved = await asyncComponent(AvatarName, {
      actionsLabel: "actionsLabel",
      role: "role",
    });

    const { container } = render(<AvatarResolved />);

    fetch("https://api.github.com/users/leandroaugusto");

    const avatar = screen.getByTestId("avatar-name");

    expect(avatar).toBeInTheDocument();
    expect(container).toMatchSnapshot();
  });

  it("should render default username and bio", async () => {
    global.fetch = jest.fn(() =>
      Promise.resolve({
        json: () => Promise.resolve({}),
      })
    ) as jest.Mock;

    const AvatarResolved = await asyncComponent(AvatarName, {
      actionsLabel: "actionsLabel",
      role: "role",
    });

    render(<AvatarResolved />);
    const heading1 = screen.getByTestId("username");
    const heading2 = screen.getByTestId("bio");

    const res = await fetch("https://api.github.com/users/usernotfound");
    const user = await res.json();

    waitFor(() => {
      expect(user.name).toEqual(undefined);
      expect(user.bio).toEqual(undefined);
      expect(heading1).toHaveTextContent("Leandro Augusto da Costa");
      expect(heading2).toHaveTextContent("role");
    });
  });

  it("should render API username and bio", async () => {
    const AvatarResolved = await asyncComponent(AvatarName, {
      actionsLabel: "actionsLabel",
      role: "role",
    });

    render(<AvatarResolved />);

    const res = await fetch("https://api.github.com/users/leandroaugusto");
    const user = await res.json();

    waitFor(() => {
      expect(user.name).toEqual("Leandro Costa");
      expect(user.bio).toEqual("bio");
    });
  });
});
