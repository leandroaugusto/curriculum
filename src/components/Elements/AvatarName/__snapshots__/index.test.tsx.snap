// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AvatarName renders the AvatarName element 1`] = `
<div>
  <div
    class="text-gray-800 ml-6"
    data-testid="avatar-name"
  >
    <h1
      class="text-2xl md:text-4xl md:mb-2"
      data-testid="username"
    >
      Leandro Costa
    </h1>
    <h2
      class="text-sm md:text-lg"
      data-testid="bio"
    >
      bio
    </h2>
    <button
      class="rounded-md bg-slate-500 text-slate-100 px-3 py-1 uppercase text-xs md:text-sm font-bold mt-4 print:hidden"
      data-testid="scroll-to-bottom-button"
    >
      actionsLabel
    </button>
  </div>
</div>
`;
